### ISSUE TITLE
Ambiguity in get function

### ISSUE THREAD

**Issue by tknopp** [JuliaLang/IniFile.jl#16]
The following definitions are ambiguous
``` 
 get(inifile::IniFile.Inifile, section::AbstractString, key::AbstractString) 
 get(inifile::IniFile.Inifile, key::AbstractString, notfound)
```
if the default value `notfound` is of type `AbstractString`

**Comment by hustf** [comment]
But is it ever? Does it produce ambiguity warnings?

**Comment by tknopp** [comment]
No, of course not always. Actually this produced a very subtle bug when changing
`get(inifile, key, 1.0)` to `get(inifile, key, "1.0")`
API-wise its not clear to me why 
`get(inifile::IniFile.Inifile, key::AbstractString, notfound)`
should be accessible for any type but a String.

So this bug was meant to report this API issue.
